src/Storages/ReadInOrderOptimizer.cpp

```diff
@@ -12,6 +12,27 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+static std::optional<Names> tryGetSortingKeyColumns(const StoragePtr & storage)
+{
+    Names sorting_key_columns;
+    if (const auto * merge_tree = dynamic_cast<const MergeTreeData *>(storage.get()))
```

Comment:
Now `getSortingKeyColumns` method is a part of `IStorage` interface and it returns empty list of columns if storage doesn't support sorting key. So, dynamic_casts are unneeded.